### PR TITLE
Add nested kvm basic test

### PR DESCRIPTION
**Pull request by lizzha**
1. Add nested kvm basic test
2. Fix issue in detecting distro in azuremodules.sh, also add debian support
3. Test pass against Ubuntu 16.04, SLES 12 SP3, CentOS 7.5, RHEL 7.4, Orace Linux 7.4, OpenSuSE Leap 42.3, Debian 9

### CODE REVIEW

File: XML/TestCases/NestedKvmTests.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<TestCases>
+	<test>
+		<TestName>NESTED-KVM-BASIC-VERIFICATION</TestName>
+		<testScript>nested_kvm_basic_test.sh</testScript>
+		<PowershellScript>NESTED-KVM-BASIC-VERIFICATION.ps1</PowershellScript>
+		<files>.\Testscripts\Linux\azuremodules.sh,.\Testscripts\Linux\nested_kvm_basic_test.sh</files>
+		<setupType>E4v3</setupType>
+		<TestParameters>
+			<!--The nested image should be qcow2 format, and accessible from a public URL-->
+			<param>NestedImageUrl=REPLACE_THIS</param>
```
Comment (lizzha) [inline XML/TestCases/NestedKvmTests.xml]: Should I place a real nested image url/user/password here?

File: XML/TestCases/NestedKvmTests.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<TestCases>
+	<test>
+		<TestName>NESTED-KVM-BASIC-VERIFICATION</TestName>
+		<testScript>nested_kvm_basic_test.sh</testScript>
+		<PowershellScript>NESTED-KVM-BASIC-VERIFICATION.ps1</PowershellScript>
+		<files>.\Testscripts\Linux\azuremodules.sh,.\Testscripts\Linux\nested_kvm_basic_test.sh</files>
+		<setupType>E4v3</setupType>
```
Comment (iamshital) [inline XML/TestCases/NestedKvmTests.xml]: Hi Liz, You can change the setupType to SingelVM and override the size to E4_v3

`<setupType>SingleVM</setupType>`
`<OverrideVMSize>Standard_E4_v3</OverrideVMSize>`

This way, there is no need to add another VM configuration.
Let me know.

File: XML/TestCases/NestedKvmTests.xml (hunk 1 above)
Comment (iamshital) [reply XML/TestCases/NestedKvmTests.xml]: Hi Liz, I'm working pushing test parameters dynamically. I'll be done by this weekend. Once it is done, we no longer need to expose any sensitive information. Give me a day or two.

File: XML/TestCases/NestedKvmTests.xml (hunk 2 above)
Comment (lizzha) [reply XML/TestCases/NestedKvmTests.xml]: Cool

File: Testscripts/Linux/azuremodules.sh [hunk 3]
```diff
@@ -54,43 +54,46 @@ function detect_linux_ditribution_version()
...
 {
-    local  linux_ditribution=`cat /etc/*release*|sed 's/"//g'|grep "^ID="| sed 's/ID=//'`
+    local linux_ditribution=`cat /etc/*release*|sed 's/"//g'|grep "^ID="| sed 's/ID=//'`
     local temp_text=`cat /etc/*release*`
-    if [ "$linux_ditribution" == "" ]
-    then
+    if [ "$linux_ditribution" == "" ]; then
         if echo "$temp_text" | grep -qi "ol"; then
-            linux_ditribution='Oracle'
+            linux_ditribution='oracle'
```
Comment (nicepat) [inline Testscripts/Linux/azuremodules.sh]: Are we using all distro name in lower case inside script? Please confirm.

File: Testscripts/Linux/azuremodules.sh [hunk 4]
```diff
@@ -162,6 +165,27 @@ function install_package ()
 	done
 }
 
+function install_epel ()
+{
+    wget https://dl.fedoraproject.org/pub/epel/7/x86_64/Packages/e/epel-release-7-11.noarch.rpm
+    rpm -i epel-release-7-11.noarch.rpm
```
Comment (nicepat) [inline Testscripts/Linux/azuremodules.sh]: Check out the result of rpm installation and logging the output success or fail.

File: Testscripts/Linux/azuremodules.sh [hunk 5]
```diff
@@ -162,6 +165,27 @@ function install_package ()
...
+    check_exit_status "install_epel"
+}
+
+function install_sshpass ()
+{
+    if [[ `which sshpass` == "" ]]; then
```
Comment (nicepat) [inline Testscripts/Linux/azuremodules.sh]: Suggest to use $? for command execution result.

File: Testscripts/Linux/azuremodules.sh (hunk 3 above)
Comment (lizzha) [reply Testscripts/Linux/azuremodules.sh]: the function "detect_linux_ditribution" is not used in any other script, except in this PR

File: Testscripts/Linux/nested_kvm_basic_test.sh [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+# See the Apache Version 2.0 License for specific language governing
+# permissions and limitations under the License.
+#
+#######################################################################
+
+#######################################################################
+#
+#
+#
+# Description:
```
Comment (nicepat) [inline Testscripts/Linux/nested_kvm_basic_test.sh]: please add brief description of this script and how it works. and parameter info also.

File: Testscripts/Linux/azuremodules.sh (hunk 4 above)
Comment (lizzha) [reply Testscripts/Linux/azuremodules.sh]: it's done in the next line: 
check_exit_status "install_epel"

File: Testscripts/Linux/nested_kvm_basic_test.sh (hunk 6 above)
Comment (lizzha) [reply Testscripts/Linux/nested_kvm_basic_test.sh]: sure

File: Testscripts/Linux/azuremodules.sh (hunk 5 above)
Comment (lizzha) [reply Testscripts/Linux/azuremodules.sh]: will update

File: XML/TestCases/NestedKvmTests.xml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+		<setupType>SingleVM</setupType>
+		<OverrideVMSize>Standard_E4_v3</OverrideVMSize>
+		<TestParameters>
+			<!--The nested image should be qcow2 format, and accessible from a public URL-->
+			<param>NestedImageUrl=REPLACE_THIS</param>
+			<param>NestedUser=REPLACE_THIS</param>
+			<param>NestedUserPassword=REPLACE_THIS</param>
+			<param>HostFwdPort=60022</param>
+		</TestParameters>
+		<Platform>Azure</Platform>
```
[the remark this answers is not in the document]
Comment (lizzha) [reply XML/TestCases/NestedKvmTests.xml]: yes

[the remark this answers is not in the document]
Comment (lizzha) [reply XML/TestCases/NestedKvmTests.xml]: Currently Dv3 and Ev3 sizes in Azure support nested VM

File: Testscripts/Linux/nested_kvm_basic_test.sh [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+    fi
+    if [[ `which qemu-system-x86_64` == "" ]]; then
+        LogMsg "Cannot find qemu-system-x86_64"
+        UpdateTestState $ICA_TESTFAILED
+        exit 1
+    fi
+
+    LogMsg "Start the nested VM"
+    qemu-system-x86_64 -smp 2 -m 2048 -hda $ImageName -display none -device e1000,netdev=user.0 -netdev user,id=user.0,hostfwd=tcp::$HostFwdPort-:22 -enable-kvm &
+    sleep 30
```
[the remark this answers is not in the document]
Comment (lizzha) [reply Testscripts/Linux/nested_kvm_basic_test.sh]: tested we can change it to shorter

File: XML/TestCases/NestedKvmTests.xml [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+		<Category>Functional</Category>
+		<Area>Nested</Area>
```
Comment (nicepat) [inline XML/TestCases/NestedKvmTests.xml]: Can you update the line #18 with 'KVM'? I think this is correct path in category and would like to go detail in low level in Area. Azure -> Functional -> KVM -> <Region>. Also please add 'kvm' in tag like <Tags>nested,kvm</Tags>

Comment (lizzha) [reply XML/TestCases/NestedKvmTests.xml]: Hi Ju, I think "Nested" is the correct area, as the feature is nested VM. I also had discussed it with Simon. I will add 'kvm' in tag.

Comment (juhlee-microsoft) [reply XML/TestCases/NestedKvmTests.xml]: If Simon agrees with it, I am ok.

**Pull request merged**